pkg/transforms/batch.go

```diff
@@ -51,7 +51,6 @@ func NewBatchByCount(batchThreshold int) (*BatchConfig, error) {
 		batchThreshold: batchThreshold,
 		batchMode:      BatchByCountOnly, //Default to CountAndTime
 	}
-	config.done = make(chan bool, 1)
 
 	return &config, nil
```

Comment:
Where does the channel get created now?

Reply:
NM

Reply:
it doesnt, that was the problem it getting created and no listeners